tests/test_runlib.py

```diff
@@ -501,6 +503,47 @@ def test_in_toto_run_compare_dumped_with_returned_link(self):
         FILENAME_FORMAT.format(step_name=self.step_name, keyid=self.key["keyid"]))
     self.assertEqual(repr(link), repr(link_dump))
 
+  def test_in_toto_run_with_metadata_directory(self):
+    """Successfully run with metadata directory,
+     compare dumped link is equal to returned link"""
+    try:
+      tmp_dir = tempfile.mkdtemp()
+      link = in_toto_run(self.step_name, [self.test_artifact],
+          [self.test_artifact], ["python", "--version"], True, self.key,
+          metadata_directory=tmp_dir)
+      file_path = os.path.join(tmp_dir, self.step_name)
+      link_dump = Metablock.load(
+          FILENAME_FORMAT.format(step_name=file_path, keyid=self.key["keyid"]))
+      self.assertEqual(repr(link), repr(link_dump))
+
+    #clean up
+    finally:
+      os.remove(FILENAME_FORMAT.format(step_name=file_path, keyid=self.key["keyid"]))
+      os.rmdir(tmp_dir)
+
+  def test_in_toto_run_compare_with_and_without_metadata_directory(self):
+    """Successfully run with and without metadata directory,
+     compare the signed is equal"""
+    try:
+      tmp_dir = tempfile.mkdtemp()
+      in_toto_run(self.step_name, [self.test_artifact], [self.test_artifact],
+          ["python", "--version"], True, self.key, metadata_directory=tmp_dir)
+      file_path = os.path.join(tmp_dir, self.step_name)
```

Comment:
This works, no doubt about that, but let's avoid confusion by keeping `FILENAME_FORMAT.format()` with just the step name, and passing that as a whole to `os.path.join`.

```
file_path = os.path.join(tmp_dir, FILENAME_FORMAT.format(step_name=self.step_name, keyid=self.key["keyid"]))
link_dump_with_md = Metablock.load(file_path)
```

What do you think?